include/ipr/interface

```diff
@@ -1536,6 +1565,14 @@ namespace ipr {
       virtual Optional<Expr_list> initializer() const = 0;
    };
 
+                                   // -- Coerce --
+   // Implicit conversions that are not promotions or demotions
+   struct Coerce : Category<conditional_cat> {
```

Comment:
This could also be implemented with a Classic-Binary-Expr node but that didn't seem fit the description of that base class.

Reply:
It is indeed a Classic-Binary-Expr: the reason is, as you suspected, a coercion can be a standard conversion (therefore builtin) or implemented in code by either a converting constructor or a conversion function.